tests/unit/src/main/scala/tests/BaseTypeDefinitionLspSuite.scala

```diff
@@ -0,0 +1,111 @@
+package tests
+
+import scala.meta.internal.metals.TextEdits
+import org.eclipse.{lsp4j => l}
+import scala.concurrent.Future
+import scala.meta.inputs.Input
+import scala.meta.inputs.Position
+import scala.meta.internal.mtags.MtagsEnrichments._
+
+class BaseTypeDefinitionLspSuite extends BaseLspSuite("typeDefinition") {
+  def check(
+      name: String
+  )(query: String, expectedLocs: List[String] = Nil): Unit = {
+    testAsync(name) {
+      cleanWorkspace()
+      val code =
+        query
+          .replaceAll("(@@)?(<<)?(>>)?", "")
+          .replaceAll("""\/\*[^\*]+\*\/""", "")
+      val files = query.lines
+        .filter(_.startsWith("/"))
+        .filter(_.filter(!_.isWhitespace) != "/metals.json")
+        .map(_.stripPrefix("/"))
+
+      for {
+        _ <- server.initialize(
+          s"""${code.trim}""".stripMargin
+        )
+        _ <- Future.sequence(files.map(server.didOpen))
+        expected = expectedLocs.flatMap(
+          TestingServer.locationFromString(_, workspace)
+        )
+        _ <- server.assertTypeDefinition(
+          queryStr = query,
+          expectedLocs = expected,
+          root = workspace
+        )
+        _ = assertNoDiagnostics()
+        _ = if (!server.server.isInitialized.get())
+          fail("Build tool not initialized")
+      } yield ()
+    }
+  }
+
+  def prepareDefinition(original: String): String =
+    original.replaceAll("(<<)?(>>)?", "")
+
+  def locationsToCode(
```

Comment:
This is already used in the PCSuite I think? Can we reuse the same code?

Reply:
I think I removed duplicates of it.